File: Javascript/66.plus-one.js
/*
 * @lc app=leetcode id=66 lang=javascript
 *
 * [66] Plus One
 */

// @lc code=start
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    const addedD = [];
    let stack = 1;
    digits.reverse().forEach((v, i) => {
        let added = v + stack;
        stack = 0;
        if (added >= 10) {
            stack = 1;
            added -= 10;
        }
        addedD.push(added);
    });
    if (stack > 0) {
        addedD.push(stack);
    }
    return addedD.reverse();
};
// @lc code=end

File: Ruby/66.plus-one.rb
#
# @lc app=leetcode id=66 lang=ruby
#
# [66] Plus One
#

# @lc code=start
# @param {Integer[]} digits
# @return {Integer[]}
def plus_one(digits)
    digits[digits.size - 1] += 1

    isCarried = false
    (digits.size - 1).downto(0) { |i|
        if isCarried
            digits[i] += 1
            isCarried = false
        end

        if digits[i] > 9
            digits[i] -= 10
            isCarried = true
            next
        end
    }

    if isCarried
        digits.unshift(1)
    end

    return digits
end
# @lc code=end
